src/components/PageModule/index.tsx

```diff
@@ -1,13 +1,21 @@
 import React from 'react';
 
-import { PageModule as Props, DeviceSizeImgHeight } from '../../types';
+import { DeviceSizeImgHeight } from '../../types';
 import Header from './components/Header';
 import Carousel from './components/Carousel';
 import Grid from './components/Grid';
 import { Devider, CardTypeImgHeight } from './styles';
+import config from './config';
+
+interface Props {
+  title?: string;
+  items: string[];
+  type: string;
+}
 
 const PageModule = (props: Props) => {
-  const { title, cardType, component, imgHeight: customImgHeight } = props;
+  const { title, type, items } = props;
+  const { cardType, component, col, imgHeight: customImgHeight } = config[type];
```

Comment:
```suggestion
  const { component, ...rest } = config[type];
```
Может лучше так?
И ниже
```
{component === 'Carousel' && <Carousel {...rest} />}
{component === 'Grid' && <Grid {...rest} />}
```

Reply:
Ну и если компонент может быть только 2х типов, то я бы сделал тернарником
```
const Component = component === 'Carousel' ? Carousel : Grid;
...
<Component {...rest} />
```